cli/dcoscli/node/main.py

```diff
@@ -148,3 +165,61 @@ def _mesos_files(master, slave_id):
         slave = mesos.get_master().slave(slave_id)
         files.append(mesos.MesosFile('/slave/log', slave=slave))
     return files
+
+
+def _ssh(master, slave, option, config_file, user):
+    """SSH into a DCOS node.  Since only the masters are definitely
+    publicly available, we first ssh into an arbitrary master, then
+    hop to the desired node.
+
+    :param master: True if we ssh into the master
+    :type master: bool
+    :param slave: The slave ID to ssh into
+    :type slave: str
+    :param option: SSH option
+    :type option: str
+    :param config_file: SSH config file
+    :type config_file: str
+    :param user: SSH user
+    :type user: str
+    :rtype: int
+    :returns: process return code
+    """
+    if not os.environ.get('SSH_AUTH_SOCK'):
+        raise DCOSException(
+            "There is no SSH_AUTH_SOCK env variable, which likely means you " +
+            "aren't running `ssh-agent`.  `dcos node ssh` depends on " +
+            "`ssh-agent` so we can safely use your private key to hop " +
+            "between nodes in your cluster.  Please run `ssh-agent`, " +
+            "then add your private key with `ssh-add`.")
+
+    master_public_ip = mesos.DCOSClient().metadata()['PUBLIC_IPV4']
+    ssh_options = ' '.join('-o {}'.format(opt) for opt in option)
+
+    if config_file:
+        ssh_config = '-F {}'.format(config_file)
+    else:
+        ssh_config = ''
+
+    if master:
+        host = 'leader.mesos'
+    else:
+        summary = mesos.DCOSClient().get_state_summary()
+        slave_obj = None
+        for slave_ in summary['slaves']:
+            if slave_['id'] == slave:
+                slave_obj = slave_
+                break
```

Comment:
I was thinking we could just do this http://www.goodmami.org/2013/01/30/Getting-only-the-first-match-in-a-list-comprehension.html